sdk/core/azure-core/src/main/java/com/azure/core/util/jsonpatch/JsonPatchDocument.java

```diff
@@ -236,46 +237,56 @@ public JsonPatchDocument appendTest(String path, Object value) {
      *
      * <p><strong>Code Samples</strong></p>
      *
-     * {@codesnippet com.azure.core.experimental.jsonpatch.JsonPatchDocument.appendTestRaw#String-String}
+     * {@codesnippet com.azure.core.util.jsonpatch.JsonPatchDocument.appendTestRaw#String-String}
      *
      * @param path The path to test.
      * @param rawJson The raw JSON value that will be used to test against.
      * @return The updated JsonPatchDocument object.
      * @throws NullPointerException If {@code path} is null.
      */
     public JsonPatchDocument appendTestRaw(String path, String rawJson) {
-        return appendTestInternal(path, () -> Optional.ofNullable(rawJson));
+        return appendTestInternal(path, () -> Option.of(rawJson));
     }
 
-    private JsonPatchDocument appendTestInternal(String path, Supplier<Optional<String>> jsonSupplier) {
+    private JsonPatchDocument appendTestInternal(String path, Supplier<Option<String>> jsonSupplier) {
         Objects.requireNonNull(path, "'path' cannot be null.");
 
         return appendOperation(JsonPatchOperationKind.TEST, null, path, jsonSupplier.get());
     }
 
-    private Optional<String> serializeValue(Object value) {
+    private Option<String> serializeValue(Object value) {
         if (value == null) {
-            return Optional.empty();
+            return Option.empty();
         }
 
         String rawValue;
         try {
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
             if (serializer == null) {
-                rawValue = MAPPER.writeValueAsString(value);
+                if (DEFAULT_SERIALIZER == null) {
```

Comment:
not sure but probably static + DCL is safe in latest version of java, I remember spot bug alerting DCL in the past. jfyi, there is also an init-on-demand-holder [pattern](https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom) for these cases which won't require an explicit lock.

Reply:
Yeah, I agree that if we want lazy loading use holder pattern. DCL with static might not be safe. If this is a constant then, it should be initialized by the classloader in a static block.

Reply:
Made the static violatile.

Reply:
with volatile, happens-before is guaranteed, so DCL should be good now.